tests/TestCase/PaginationResultTest.php

```diff
@@ -15,6 +16,21 @@
 
 class PaginationResultTest extends TestCase
 {
+    public function setUp(): void
+    {
+        set_error_handler(
+            static function ($errno, $errstr) {
+                throw new \Exception($errstr, $errno);
```

Comment:
How about throwing [`ErrorException`](https://www.php.net/manual/en/class.errorexception.php), and expecting `\ErrorException::class` in `testUndefinedProperties` instead?